allennlp/tests/data/dataset_readers/multiprocess_dataset_reader_test.py

```diff
@@ -72,6 +75,53 @@ def test_multiprocess_read(self):
         assert counts[("snakes", "are", "animals", ".", "N", "V", "N", "N")] == 100
         assert counts[("birds", "are", "animals", ".", "N", "V", "N", "N")] == 100
 
+    def test_multiprocess_read_partial_does_not_hang(self):
+        reader = MultiprocessDatasetReader(base_reader=self.base_reader, num_workers=4, output_queue_size=10)
+
+        all_instances = []
+
+        # Half of 100 files * 4 sentences / file
+        i = 0
+        for instance in reader.read(self.identical_files_glob):
+            if i == 200:
```

Comment:
I would probably just add a comment here, along the lines of "i == 200 means this is the 201st instance, which is too many, so break out and fail the test"